### PR TITLE
Added Ability to Report Interior Moveable Insulation Inside Face Surface Temperature

### LINKED ISSUE #4432: Surface Inside Temperature not clear with Movable Insulation - possible error in calculation
Helpdesk Ticket 7095

Code appears to be reporting the base surface inside temp. Checking with Brent and Rick to confirm.
 Michael J. Witte on 22 February 2013 03:40 AM

User was never answered - been six months - needs a CR. Reference e-mail from RKS to MJW on FEb 25, 2013
 Michael J. Witte on 20 August 2013 05:45 PM

Is this addressed in Rick's fix for interior movable insulation?
 Linda Lawrie on 16 October 2013 07:52 PM

From RKS on 25 Feb 2013 - Yeah, well, it's "temporary" but then movable insulation uses it to differentiate between base and movable insulation surface temperature. My guess is that only a handful of people know this (a majority on this email). So, Mike, I think your interpretation is correct and the report probably needs correcting for the limited cases when these two temperatures are not identical. Or course, this is going from memory, but what Mike writes sounds like a reasonable explanation. Sorry that it takes a while to get a response from me at the moment. (reference e-mail thread for more details when posting new issue)
 Michael J. Witte on 07 May 2014 04:29 PM (Edited by Michael J. Witte on 07 May 2014 04:29 PM)

Helia Taheri
User
Posted on: 15 February 2013 09:09 AM
Hi again Dear Linda,

 I have another question:
 When I have an inside movable insulation and I calculate the Surface Inside Temperature, the software will give me the temperature of the main surface or the temperature of the inside movable insulation? 

 Thanks
 Helia

### PR DESCRIPTION
**Pull request by RKStrand**
Pull request overview
---------------------
When choosing interior movable insulation, there is a temperature calculated at the interface between the zone air and the interior moveable insulation as well as a temperature at the interface between the interior moveable insulation and the actual surface itself.  When the user requests the inside face temperature, they receive the temperature at the interface between the moveable insulation and the surface.  This work added the capability to also report the temperature at the interface between the zone air and the moveable insulation.  This work resolves #4432
### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue #4432 

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add to input rules file for interfaces
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Required documentation updates?
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by rraustad** [comment]
New report variable added, no structural change to tables.

File: tst/EnergyPlus/unit/HeatBalanceSurfaceManager.unit.cc [hunk 1]
```diff
@@ -2488,4 +2488,52 @@ TEST_F(EnergyPlusFixture, HeatBalanceSurfaceManager_TestSurfTempCalcHeatBalanceA
...
+    DataHeatBalSurface::TempSurfIn.allocate(DataSurfaces::TotSurfaces);
+    DataHeatBalSurface::TempSurfInTmp.allocate(DataSurfaces::TotSurfaces);
+    DataHeatBalSurface::TempSurfInMovInsRep.allocate(DataSurfaces::TotSurfaces);
+
+    // Test 1 Data: Surface does NOT have movable insulation
+    DataSurfaces::Surface(1).MaterialMovInsulInt = 0;   // No material means no movable insulation
+    DataSurfaces::Surface(1).SchedMovInsulInt = 0;      // Schedule index of zero returns zero value (not scheduled)
+    DataHeatBalSurface::TempSurfIn(1) = 23.0;
+    DataHeatBalSurface::TempSurfInTmp(1) = 12.3;
+    DataHeatBalSurface::TempSurfInMovInsRep(1) = 1.23;
```
Comment (rraustad) [inline tst/EnergyPlus/unit/HeatBalanceSurfaceManager.unit.cc:2512]: Clear test of TempSurfIn* choice by setting the report var to an invalid value before calling HeatBalanceSurfaceManager::ReportIntMovInsInsideSurfTemp

File: tst/EnergyPlus/unit/HeatBalanceSurfaceManager.unit.cc [hunk 2]
```diff
@@ -2488,4 +2488,52 @@ TEST_F(EnergyPlusFixture, HeatBalanceSurfaceManager_TestSurfTempCalcHeatBalanceA
...
+    DataHeatBalSurface::TempSurfIn(3) = 12.3;
+    DataHeatBalSurface::TempSurfInTmp(3) = 1.23;
+    DataHeatBalSurface::TempSurfInMovInsRep(3) = -9999.9;
+    ExpectedResult3 = 1.23;                             // TempSurfInMovInsRep should be set to TempSurfInTmp
+
+    // Now call the subroutine which will run all of the test cases at once and then make the comparisons
+    HeatBalanceSurfaceManager::ReportIntMovInsInsideSurfTemp();
+    EXPECT_NEAR(DataHeatBalSurface::TempSurfInMovInsRep(1),ExpectedResult1,0.00001);
+    EXPECT_NEAR(DataHeatBalSurface::TempSurfInMovInsRep(2),ExpectedResult2,0.00001);
+    EXPECT_NEAR(DataHeatBalSurface::TempSurfInMovInsRep(3),ExpectedResult3,0.00001);
```
Comment (rraustad) [inline tst/EnergyPlus/unit/HeatBalanceSurfaceManager.unit.cc:2535]: Very clear unit test

File: src/EnergyPlus/HeatBalanceSurfaceManager.cc [hunk 3]
```diff
@@ -5105,6 +5109,18 @@ namespace HeatBalanceSurfaceManager {
...
+        int SurfNum;
+        TempSurfInMovInsRep = TempSurfIn;
+        for (SurfNum = 1; SurfNum <= TotSurfaces; ++SurfNum) {
+            if (Surface(SurfNum).MaterialMovInsulInt > 0) {
+                if (GetCurrentScheduleValue(Surface(SurfNum).SchedMovInsulInt) > 0.0) {
+                    TempSurfInMovInsRep(SurfNum) = TempSurfInTmp(SurfNum);
+                }
+            }
+        }
+    }
```
Comment (rraustad) [inline src/EnergyPlus/HeatBalanceSurfaceManager.cc:5123]: Code already sets moveable insulation interior surface temp to TempSurInTemp but never reported it (i.e., you won't find TempSurfInTmp() = something in this Git review, must go to code). This function sets the new report variable.

**Comment by rraustad** [comment]
@Myoldmopar these are false diff's based on changes in other branches. I expect 0 diff's with this change. What is the course of action in this case where diff's are present and none are expected? I'm going to pull the branch and build/read the docs and run the example file. Should I merge develop and push to get fresh CI results?

**Comment by rraustad** [comment]
@RKStrand are you finished here? I am ready for a final review.

**Comment by RKStrand** [comment]
@rraustad Yes, I am finished with this.  Feel free to proceed to the final review.

**Comment by rraustad** [comment]
- Final code and unit test review
 - Compiled and reviewed Eng. Ref and IO Ref., equations look good, no new figures to add to CMake.
 - Ran sequential unit tests in release mode without failure (after pulling develop)
 - Ran example file, no warnings in err file. Same for annual run.
 - Verified new report variable added to existing example file (MovableIntInsulationSimple.idf)
 - Plotted new report variable for design days

image

**Comment by rraustad** [comment]
No need to run CI again, first few CI results show expected diff's.

**Pull request merged**